### PR TITLE
Add new column for Server Processes UI

### PR DESCRIPTION
**Pull request by vuon9**
## Changes:
- Add new column for Server Processes UI

## Closes following issues:
- Closes: #2061 

## Tested:
- Processors:
  - [ ] Intel
  - [x] Apple Silicon
- macOS Versions:
  - [ ] 10.13.x (High Sierra)
  - [ ] 10.14.x (Mojave)
  - [ ] 10.15.x (Catalina)
  - [ ] 11.x (Big Sur)
  - [ ] 12.x (Monterey)
  - [ ] 13.x (Ventura)
  - [x] 14.x (Sonoma)
- Localizations:
  - [x] English
  - [ ] Spanish
  - [ ] Other (please specify)
- Xcode Version:
  
## Screenshots:

image

### CODE REVIEW

File: Source/Controllers/SubviewControllers/SPProcessListController.m [hunk 1]
```diff
@@ -686,7 +688,8 @@ - (void)_updateServerProcessesFilterForFilterString:(NSString *)filterString
 			([[process objectForKey:@"Command"] rangeOfString:filterString options:NSCaseInsensitiveSearch].location != NSNotFound) ||
 			((![[process objectForKey:@"Time"] isNSNull]) && ([[[process objectForKey:@"Time"] stringValue] rangeOfString:filterString options:NSCaseInsensitiveSearch].location != NSNotFound)) ||
 			((![[process objectForKey:@"State"] isNSNull]) && ([[process objectForKey:@"State"] rangeOfString:filterString options:NSCaseInsensitiveSearch].location != NSNotFound)) ||
-			((![[process objectForKey:@"Info"] isNSNull]) && ([[process objectForKey:@"Info"] rangeOfString:filterString options:NSCaseInsensitiveSearch].location != NSNotFound)))
+			((![[process objectForKey:@"Info"] isNSNull]) && ([[process objectForKey:@"Info"] rangeOfString:filterString options:NSCaseInsensitiveSearch].location != NSNotFound)) ||
+      ((![[process objectForKey:@"Progress"] isNSNull]) && ([[process objectForKey:@"Progress"] rangeOfString:filterString options:NSCaseInsensitiveSearch].location != NSNotFound)))
```
Comment (Jason-Morcos) [inline Source/Controllers/SubviewControllers/SPProcessListController.m:692]: Just to double check - how does this work/look if using MySQL that doesn't include Progress in the response? Should there perhaps be a guard to fully hide that column if not defined in the response to keep from confusing non-MariaDB folks?

**Review by Jason-Morcos — COMMENTED** [review]
Looks good! Just one question to resolve about how this will work for non-MariaDB users

File: Source/Controllers/SubviewControllers/SPProcessListController.m (hunk 1 above)
Comment (vuon9) [reply Source/Controllers/SubviewControllers/SPProcessListController.m:692]: Will look at it. Thanks!

**Comment by vuon9** [comment]
I've changed and tested on both mysql and mariadb, pls help to review again. Thanks!
image
image
image

File: Source/Controllers/SubviewControllers/SPProcessListController.m [hunk 2]
```diff
@@ -244,15 +247,16 @@ - (IBAction)saveServerProcesses:(id)sender
...
                                            [process objectForKey:@"Id"],
                                            [process objectForKey:@"User"],
                                            [process objectForKey:@"Host"],
                                            [process objectForKey:@"db"],
                                            [process objectForKey:@"Command"],
                                            [process objectForKey:@"Time"],
                                            [process objectForKey:@"State"],
-                                           [process objectForKey:@"Info"]];
+                                           [process objectForKey:@"Info"],
+                                           [process objectForKey:@"Progress"]];
```
Comment (Jason-Morcos) [inline Source/Controllers/SubviewControllers/SPProcessListController.m:259]: I noticed this isn't a safeObjectFroKey accessing here or in the other place in this controller - can we confirm it won't crash for MySQL responses?

**Review by Jason-Morcos — COMMENTED** [review]
One final thing to check - to make sure it doesn't crash for MySQL! As long as this doesn't crash there, it looks great to me!

**Review by Kaspik — APPROVED** [review]

**Comment by vuon9** [comment]
@Jason-Morcos I don't quite understand because I shown the screenshots of 2 versions of MySQL to prove that.

File: Source/Controllers/SubviewControllers/SPProcessListController.m (hunk 2 above)
Comment (vuon9) [reply Source/Controllers/SubviewControllers/SPProcessListController.m:259]: I already tested on MariaDB/MySql versions and have no crash issues.

Comment (Jason-Morcos) [reply Source/Controllers/SubviewControllers/SPProcessListController.m:259]: I must have missed that, my bad! Nice work on this one!

**Comment by kimegede** [comment]
Could this be related to the bug I'm having, where filtering stopped working?
https://github.com/Sequel-Ace/Sequel-Ace/issues/2153

**Pull request merged**